### PR TITLE
Axis refactor v7 - Legend interpolation, Try to separate axis generator functions

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/chart/generator/axis.h [hunk 1]
```diff
@@ -53,13 +55,17 @@ struct MeasureAxis
 {
 	::Anim::Interpolated<bool> enabled{false};
 	Math::Range<double> range = Math::Range<double>::Raw(0, 1);
+	std::string series;
 	::Anim::String unit;
```
Comment (schaumb) [inline src/chart/generator/axis.h:59]: Unit not anim, axis not needed

File: src/chart/generator/axis.h [hunk 2]
```diff
@@ -81,17 +87,22 @@ struct DimensionAxis
 		Math::Range<double> range;
 		double value;
 		::Anim::Interpolated<ColorBase> colorBase;
-		::Anim::String label;
-		std::string categoryValue;
+		::Anim::Interpolated<bool> label;
```
Comment (schaumb) [inline src/chart/generator/axis.h:90]: Not interpolated, but removed if no label needed.

File: src/chart/generator/axis.h (hunk 1 above)
Comment (schaumb) [reply src/chart/generator/axis.h:59]: Unit is anim. (%)

File: src/chart/generator/axis.h (hunk 2 above)
Comment (schaumb) [reply src/chart/generator/axis.h:90]: not remove the labels, because interpolation is useful anyway.

**Review by simzer — APPROVED** [review]

**Pull request merged**